docs/templates/snakemake/workflow/Snakefile

```diff
@@ -7,6 +7,9 @@ import pandas as pd
 samples = pd.read_csv(os.path.abspath(
     config["samples"])).set_index("sample", drop=False)
 
+assert isinstance(config["run_identification"], bool), f"'run_identification' must be True/False boolean, {config['run_identification']} (type {type(config['run_identification'])}) was provided"
+assert isinstance(config["run_quantification"], bool), f"'run_quantification' must be True/False boolean, {config['run_quantification']} (type {type(config['run_quantification'])}) was provided"
+assert isinstance(config["run_differential"], bool), f"'run_differential' must be True/False boolean, {config['run_differential']} (type {type(config['run_differential'])}) was provided"
```

Comment:
> Maybe add an extra step that reports missing flags to make sure that it is the desired behaviour. Something like: "INFO: The flag run_differential is missing. If this method does not support it, you can disregard this message."

Yes I think this is a great idea if we go with the remove the flag option.

> Maybe I got the comment wrong and I should not remove the whole line in the first place. In that case your assert statements would correctly catch that.


No you are absolutely correct, I think elsewhere in the README I said to remove the flag if the tool is not compatible with that challenge, so following from that the assert would fail if it's not removed. I agree I'm also not sure what the best option is (I've switched preferences as I've written this reply...), I think both are sensible and we'll have to decide and stick with one.

Some thoughts in favour of removing the flag altogether:

- Don't include a parameter that cannot be changed. (_maybe_ it could cause some confusion/room for accidental changes if retained in the config (though with an assert statement the inconvenience would be negligible))
- Less code in the Snakemake template (i.e. don't need to check a parameter that has no effect on the workflow)
- Don't 'suggest' a tool can perform a task is does not do (although enforce False with info message would cover this anyway)

Thoughts in favour of enforce False:
- If changes to challenges that mean tool is now compatible (e.g. a 'relative quantification' benchmark), the flag is still there (just need to remove enforce False assert statement and get Snakemake to track the output file)


I'm wondering whether we should make a distinction between the following:
- tool can perform a task **but does not fit our challenge specifications**
    - Enforce the flag to False (assert statement to check, perhaps an info message too?)
-  tool does not perform a given task at all
    - Remove the flag entirely (with the missing key info message you suggested)


My main worry with above suggestion is this template (and corresponding documentation) becomes convoluted and difficult to follow for new users (though at this stage of the project there won't be many of those...)

I feel like any option will be equally valid so I'm open to input/a decision. I think the final option is most comprehensive but would add a layer of complexity that we may not want at this stage...

Reply:
Discussed in a previous meeting, see comment for outcome below.